druntime/src/core/internal/gc/impl/conservative/gc.d

```diff
@@ -160,7 +160,7 @@ class ConservativeGC : GC
 
     Gcx *gcx;                   // implementation
 
-    static gcLock = shared(AlignedSpinLock)(SpinLock.Contention.lengthy);
```

Comment:
The reason for lengthy was of course the expected operation time holding the lock.
While busy waiting will have lower latency, it's wasteful and can block global process.

Reply:
On my 16-thread application (with each thread allocating class objects), I was getting only 2.5x performance boost. On debug I noticed that most of my threads were sleeping for 1ms. I think the locked threads should not sleep at all. We are anyway yielding and therefore giving a chance to the (waiting) global threads to run.

After making this fix, I am wow getting 9x performance with 16 threads. While I am not an expert in parallelism, in my view GC malloc is pretty fast to qualify for brief contention.